src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -1382,24 +1459,33 @@ struct action<struct_forward_dcl>
         }
         cleanup_guard{state};
 
-        Module& module = ctx->module();
+        auto module = ctx->modules().current();
         const std::string& struct_name = state["struct_name"];
-        if (module.has_symbol(struct_name, false))
+
+        if (struct_name.find("::") != std::string::npos)
+        {
+            return; // Cannot add a symbol with scoped name.
+        }
+
+        const std::string& name_space = module->scope();
+        const std::string scoped_struct_name = name_space.empty() ? struct_name : name_space + "::" + struct_name;
```

Comment:
This pattern repeats a lot. I would create a `std::string create_scoped_name(const std::string& plain_name)` method in `Module` class that will:
1. Throw a `runtime_error` exception when `plain_name` contains `::`.
2. Return the corresponding scoped name.

Reply:
If possible, we should also add negative tests for this.